src/Spd.Resource.Repository/PortalUser/PortalUserRepository.cs

```diff
@@ -95,7 +91,25 @@ private async Task<PortalUserResp> UpdatePortalUserAsync(UpdatePortalUserCmd c,
             identity = await _context.GetIdentityById((Guid)c.IdentityId, ct);
             _context.SetLink(portalUser, nameof(portalUser.spd_IdentityId), identity);
         }
+
+        ContactRoleCode? currentContactRoleCode = SharedMappingFuncs.GetContactRoleCode(portalUser.spd_spd_role_spd_portaluser);
+
+        if (c.ContactRoleCode != null && currentContactRoleCode != c.ContactRoleCode)
+        {
+            spd_role? currentRole = portalUser.spd_spd_role_spd_portaluser.FirstOrDefault();

```

Comment:
you get the role again here.

Reply:
Correct, but here I get it as an entity to link it with portal user entity